### PR TITLE
Use isTrashed | Closes #2298, #2297

### LINKED ISSUE #2298: Convert isArchived and isDeleted to isTrashed in PHP GraphQL Types
Any database fields for entity using "archived" and/or "deleted" need to be changed to use the term "trashed" for the GraphQL Types

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR:
- Removes `isDeleted` field from GQL types.
- Adds `isTrashed` field in place of `isDeleted`
- Adds GQL support for untrashing entities
- Updates client-side code to replace `isDeleted` and `isArchived` with `isTrashed`
- Updates GQL docs
- Closes #2298, #2297

### CODE REVIEW

File: assets/prototype/domain/shared/entities/datetimes/predicates/filters/aboveCapacity/index.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ type AboveCapacityProps = {
 };
 
 const aboveCapacity = ({ capacity, dates }: AboveCapacityProps): Datetime[] | [] => {
-	const filterFn = (date) => !date.isDeleted && capacityAtOrAbove(date, capacity);
+	const filterFn = (date: Datetime): boolean => !date.isTrashed && capacityAtOrAbove(date, capacity);
```
Comment (nicolad) [inline assets/prototype/domain/shared/entities/datetimes/predicates/filters/aboveCapacity/index.ts]: :+1:

File: assets/prototype/domain/shared/entities/tickets/predicates/isTrashed.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+/**
```
Comment (nicolad) [inline assets/prototype/domain/shared/entities/tickets/predicates/isTrashed.ts]: Looking at these 2 I am wondering if it is worth to have a common utility for both entities.
This way we will have to test just one predicate.
image

Comment (manzoorwanijk) [reply assets/prototype/domain/shared/entities/tickets/predicates/isTrashed.ts]: Would be common for all I think

Comment (tn3rb) [reply assets/prototype/domain/shared/entities/tickets/predicates/isTrashed.ts]: would be nice if we could create a Type called something like :

```ts
type TrashableEntity = Datetime | Event | Price | Ticket;
```

but I don't think you can create types that are a set of interfaces.

Comment (manzoorwanijk) [reply assets/prototype/domain/shared/entities/tickets/predicates/isTrashed.ts]: I like the second suggestion

File: core/domain/services/graphql/data/mutations/PriceMutation.php [hunk 3]
```diff
@@ -41,8 +40,12 @@ public static function prepareFields(array $input)
...
 
-        if (isset($input['isDefault'])) { // it can be false
-            $args['PRC_is_default'] = boolval($input['isDefault']);
+        if (array_key_exists('isDefault', $input)) {
+            $args['PRC_is_default'] = (bool) $input['isDefault'];
+        }
+
+        if (array_key_exists('isTrashed', $input)) {
+            $args['PRC_deleted'] = (bool) $input['isTrashed'];
         }
```
Comment (tn3rb) [inline core/domain/services/graphql/data/mutations/PriceMutation.php]: thnx for adding 👍

File: assets/prototype/domain/shared/entities/tickets/predicates/isTrashed.ts (hunk 2 above)
Comment (tn3rb) [reply assets/prototype/domain/shared/entities/tickets/predicates/isTrashed.ts]: ya i should have just deleted the first idea... was kinda just writing everything out as things popped into my mind

Comment (nicolad) [reply assets/prototype/domain/shared/entities/tickets/predicates/isTrashed.ts]: > ya i should have just deleted the first idea... was kinda just writing everything out as things popped into my mind



I like to follow the thought process :+1:

**Pull request merged**